pkgs/development/libraries/science/math/magma/generic.nix

```diff
@@ -0,0 +1,160 @@
+# Type aliases
+# Release = {
+#  version: String
+#  hash: String
+#  supportedGpuTargets: List String
+# }
+
+{ lib
+, stdenv
+, fetchurl
+, cmake
+, ninja
+, gfortran
+, libpthreadstubs
+, lapack
+, blas
+, cudaPackages
+, symlinkJoin
+, hip
+, hipblas
+, hipsparse
+, openmp
+, cudaSupport ? true
+, rocmSupport ? false
+, gpuTargets ? [ ]
+  # magmaRelease :: Release
+, magmaRelease
```

Comment:
please alphabetize

Reply:
@samuela is it okay to alphabetize all of them, or should there be some order within them?

Like, do arguments with defaults go last?

Reply:
i usually just alphabetize everything, eg with VSCode's "Sort lines ascending" command. Honestly we don't have a strict style guide for these things yet so as long as it's some sensible ordering scheme it should be fine